crates/evm/evm/src/inspectors/stack.rs

```diff
@@ -582,36 +647,15 @@ impl InspectorStack {
         };
         (InterpreterResult { result, output, gas }, address)
     }
-
-    /// Adjusts the EVM data for the inner EVM context.
-    /// Should be called on the top-level call of inner context (depth == 0 &&
-    /// self.in_inner_context) Decreases sender nonce for CALLs to keep backwards compatibility
-    /// Updates tx.origin to the value before entering inner context
-    fn adjust_evm_data_for_inner_context<DB: DatabaseExt>(
-        &mut self,
-        ecx: &mut EvmContext<&mut DB>,
-    ) {
-        let inner_context_data =
-            self.inner_context_data.as_ref().expect("should be called in inner context");
-        let sender_acc = ecx
-            .journaled_state
-            .state
-            .get_mut(&inner_context_data.sender)
-            .expect("failed to load sender");
-        if !inner_context_data.is_create {
-            sender_acc.info.nonce = inner_context_data.original_sender_nonce;
-        }
-        ecx.env.tx.caller = inner_context_data.original_origin;
-    }
 }
 
 // NOTE: `&mut DB` is required because we recurse inside of `transact_inner` and we need to use the
 // same reference to the DB, otherwise there's infinite recursion and Rust fails to instatiate this
 // implementation. This currently works because internally we only use `&mut DB` anyways, but if
 // this ever needs to be changed, this can be reverted back to using just `DB`, and instead using
 // dynamic dispatch (`&mut dyn ...`) in `transact_inner`.
```

Comment:
outdated comment, is this why we're using dynamic dispatch now in `transact_inner`? In that case we can move all the `&mut DB` to `DB`